inc/dropins/file-based-page-cache-functions.php

```diff
@@ -78,6 +78,12 @@ function sc_cache( $buffer, $flags ) {
 
 	$modified_time = time(); // Make sure modified time is consistent
 
+	// Prevent mixed content when there's an http request but the site URL uses https
+	if ( ! $_SERVER['HTTPS'] && 0 === ( strpos( get_option( 'siteurl' ), 'https' ) ) ) {
```

Comment:
I think this should be `home_url` not `siteurl`. Site URL is for the admin. It's not uncommong to have admin on HTTPS and front-end not (or vice-versa). Also, need to escape the value pulled from the DB (`esc_url`).

Reply:
Thank you for the review. I see other improvements that could be done:
- `get_home_url ()` should be used instead of `get_option( 'home' )`: possible use of the `home_url` filter
- `is_ssl()` should be used instead of `$_SERVER['HTTPS']`: native functions should be preferred
- `parse_url( $url, PHP_URL_SCHEME )` should be used instead of `0 === ( strpos( get_option( 'siteurl' ), 'https' ) )`: this is used in `get_home_url` to get the scheme

Do you agree with that?

Reply:
Yep!